### PR TITLE
Add create_ignore_already_exists to google_sourcerepo_repository

### PR DESCRIPTION
**Pull request by daniel-cit**
Adding `create_ignore_already_exists` field  to `google_sourcerepo_repository` resource 
Also adding polling and time sleep to easy eventually consistent creation of the repository

Part of  https://github.com/hashicorp/terraform-provider-google/issues/18087

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
sourcerepo: added `create_ignore_already_exists` field to `google_sourcerepo_repository` resource
```

### CODE REVIEW

**Comment by c2thorn** [comment]
Hi @daniel-cit 

Tomorrow, the Magic Modules repository is scheduled to undergo a language migration from Ruby to Go. You can view more details about this in our announcement here: https://github.com/hashicorp/terraform-provider-google/issues/19583 (or [go/mm-migration-announcement](go/mm-migration-announcement) if you are a Googler)

This open pull request may become incompatible due to most YAML and .erb files converting to Go-compatible files.

Our team (Magic Modules repository maintainers) has tooling to automatically convert changes to the new language, and we can prepare a new commit for this pull request that is compatible with the migration.

In order to push the new changes to your pull request, we would need to <ins>force push the commit to your fork's branch</ins>. Our tooling saves a backup branch before converting, so we could rollback or open a new pull request if needed. We would also work with you and the PR reviewer in the event additional changes are needed.

You also have the option to update the pull request yourself after the migration. You can view a [preview branch](https://github.com/GoogleCloudPlatform/magic-modules/tree/FEATURE-BRANCH-rewrite-switchover-preview) and [updated documentation](https://googlecloudplatform.github.io/magic-modules/reference/ruby-go-changes/) related to the migration changes.

We will take no action until we have your explicit permission to push changes to your fork's branch used for this pull request. Let me or your reviewer know if you have any further questions!

**Comment by daniel-cit** [comment]
> Hi @daniel-cit
> 
> Tomorrow, the Magic Modules repository is scheduled to undergo a language migration from Ruby to Go. You can view more details about this in our announcement here: [hashicorp/terraform-provider-google#19583](https://github.com/hashicorp/terraform-provider-google/issues/19583) (or [go/mm-migration-announcement](go/mm-migration-announcement) if you are a Googler)
> 
> This open pull request may become incompatible due to most YAML and .erb files converting to Go-compatible files.
> 
> Our team (Magic Modules repository maintainers) has tooling to automatically convert changes to the new language, and we can prepare a new commit for this pull request that is compatible with the migration.
> 
> In order to push the new changes to your pull request, we would need to force push the commit to your fork's branch. Our tooling saves a backup branch before converting, so we could rollback or open a new pull request if needed. We would also work with you and the PR reviewer in the event additional changes are needed.
> 
> You also have the option to update the pull request yourself after the migration. You can view a [preview branch](https://github.com/GoogleCloudPlatform/magic-modules/tree/FEATURE-BRANCH-rewrite-switchover-preview) and [updated documentation](https://googlecloudplatform.github.io/magic-modules/reference/ruby-go-changes/) related to the migration changes.
> 
> We will take no action until we have your explicit permission to push changes to your fork's branch used for this pull request. Let me or your reviewer know if you have any further questions!


Hi @c2thorn 

Please force push the commit generated by the tool to automatically convert changes to the new language.

Thanks for your help and guidance

File: mmv1/templates/terraform/constants/source_repo_repository.go.tmpl [hunk 1]
```diff
@@ -26,3 +26,47 @@ func resourceSourceRepoRepositoryPubSubConfigsHash(v interface{}) int {
...
+
+func resourceSourceRepoRepositoryPollRead(d *schema.ResourceData, meta interface{}) transport_tpg.PollReadFunc {
+	return func() (map[string]interface{}, error) {
+		config := meta.(*transport_tpg.Config)
+		userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+		if err != nil {
+			return nil, err
+		}
+
+		url, err := tpgresource.ReplaceVars(d, config, "{{SourceRepoBasePath}}projects/{{project}}/repos")
```
Comment (zli82016) [inline mmv1/templates/terraform/constants/source_repo_repository.go.tmpl:38]:
```suggestion
		url, err := tpgresource.ReplaceVars(d, config, "{{"{{"}}SourceRepoBasePath{{"}}"}}projects/{{"{{"}}project{{"}}"}}/repos")
```

File: mmv1/templates/terraform/custom_create/source_repo_repository.go.tmpl [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    obj["name"] = nameProp
+}
+pubsubConfigsProp, err := expandSourceRepoRepositoryPubsubConfigs(d.Get("pubsub_configs"), d, config)
+if err != nil {
+    return err
+} else if v, ok := d.GetOkExists("pubsub_configs"); !tpgresource.IsEmptyValue(reflect.ValueOf(pubsubConfigsProp)) && (ok || !reflect.DeepEqual(v, pubsubConfigsProp)) {
+    obj["pubsubConfigs"] = pubsubConfigsProp
+}
+
+url, err := tpgresource.ReplaceVars(d, config, "{{SourceRepoBasePath}}projects/{{project}}/repos")
```
Comment (zli82016) [inline mmv1/templates/terraform/custom_create/source_repo_repository.go.tmpl:20]:
```suggestion
url, err := tpgresource.ReplaceVars(d, config, "{{"{{"}}SourceRepoBasePath{{"}}"}}projects/{{"{{"}}project{{"}}"}}/repos")
```

File: mmv1/templates/terraform/custom_create/source_repo_repository.go.tmpl [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    return err
+}
+
+// We can't guarantee complete consistency even after polling,
+// so sleep for some additional time to reduce the likelihood of
+// eventual consistency failures.
+time.Sleep(10 * time.Second)
+
+// Store the ID now
+id, err := tpgresource.ReplaceVars(d, config, "projects/{{project}}/repos/{{name}}")
```
Comment (zli82016) [inline mmv1/templates/terraform/custom_create/source_repo_repository.go.tmpl:86]:
```suggestion
id, err := tpgresource.ReplaceVars(d, config, "projects/{{"{{"}}project{{"}}"}}/repos/{{"{{"}}name{{"}}"}}")
```

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**